# HG changeset patch
# User Daniel J Sebald  <daniel.sebald@ieee.org>
# Date 1232090866 -3600
#      Fri Jan 16 08:27:46 2009 +0100
# Node ID 774b44619c5c31952cd7920f6e91ab6aebc37803
# Parent  06e1667d7492cb3d130a68e5a509a571c29384ed
 Fix legend order for both horizontal and vertical string cell.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-01-16  Daniel J Sebald  <daniel.sebald@ieee.org>
+
+	* plot/legend.m: Fix legend order for both horizontal and
+	vertical string cell.
+
 2008-01-15  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/grid.m: Document handle argument.
 
 2009-01-15  Peter L. Søndergaard  <peter@sonderport.dk>
 
 	* general/nargoutchk.m: Doc fix.
 	* general/nargchk.m: Improve compatibility.  New tests.
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -160,23 +160,21 @@ function legend (varargin)
 	    endif
 	endswitch
 	nargs--;
       else
 	varargin = cellstr (arg);
 	nargs = numel (varargin);
       endif
     elseif (iscellstr (arg))
-      varargin = fliplr (arg);
+      varargin = arg;
       nargs = numel (varargin);
     else
       error ("legend: expecting argument to be a character string");
     endif
-  else
-    varargin(1:nargs) = fliplr (varargin(1:nargs));
   endif
 
   if (nargs > 0)
     have_data = false;
     for i = 1:nkids
       if (strcmp (get (kids(k), "type"), "line")
 	  || strcmp (get (kids(k), "type"), "surface")
 	  || strcmp (get (kids(k), "type"), "patch")
@@ -186,17 +184,17 @@ function legend (varargin)
       endif
     endfor
     if (! have_data)
       warning ("legend: plot data is empty; setting key labels has no effect");
     endif
   endif
 
   warned = false;
-  for i = 1:nargs
+  for i = nargs:-1:1
     arg = varargin{i};
     if (ischar (arg))
       while (k <= nkids
 	     && ! (strcmp (get (kids(k), "type"), "line")
 		   || strcmp (get (kids(k), "type"), "surface")
 		   || strcmp (get (kids(k), "type"), "patch")
 		   || strcmp (get (kids(k), "type"), "hggroup")))
 	k++;
